src/psyclone/nemo.py

```diff
@@ -549,8 +544,6 @@ def __init__(self, ast, parent=None):
                       valid_loop_types=valid_loop_types)
         # Keep a ptr to the corresponding node in the AST
         self._ast = ast
-        self._text_name = "NemoImplicitLoop"
```

Comment:
That was not intended, I think I deleted this at the end because CodeCov said it was not being used. In a more careful look I see the "NemoImplicitLoop" was hardcoded in the `__str__` method. I restored this _text_name and add the node_str method for proper coloring.

(Note that the other _text_name's deleted from the nemo.py file didn't really belong here)

Reply:
I seen there are some more hardcoded str methods that are not necessary, I fixed some in the code that I touched.